TestingHomework-Discounts/Accessors/CheckoutAccessor.cs

```diff
@@ -0,0 +1,201 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using TestingHomework.Tests.Data;
+
+namespace TestingHomework_Discounts.Accessors
+{
+    public interface ICheckoutAccessor
+
+    {
+        Cart RedeemPromo(string code, Cart cart);
+        Cart GetUserCart(Guid userId);
+        Cart SaveCart(Cart cart);
+        CartDiscountResult CalculateDiscounts(Cart cart);
+
+
+    }
+    public class CheckoutAccessor: ICheckoutAccessor
+    {
+        PromoCode_Mapper mapper = new PromoCode_Mapper();
+        Cart_Mapper cartMapper = new Cart_Mapper();
+        public Cart RedeemPromo(string code, Cart cart)
```

Comment:
I appreciate the attempt at using IDesign. Accessors are exclusively for interacting with the data store and should not contain logic. The rules for redeeming a promo (basically everything in this method except fetching the promo and saving the incremented usage) should be part of a manager or engine.

Also, this was the primary method to refactor. To split out to make testing simpler. However, it has been left as it was with the mixed responsibilities.